site/src/Librecores/ProjectRepoBundle/Util/DefaultExecutor.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Librecores\ProjectRepoBundle\Util;
+
+use Psr\Log\LoggerInterface;
+
+use Symfony\Component\Process\ProcessBuilder;
+
+/**
+ * Default implementation of ExecutorInterface
+ *
+ * Uses Symfony ProcessBuilder component to build the process.
+ *
+ * @author Amitosh Swain Mahapatra <amitosh.swain@gmail.com>
+ */
+class DefaultExecutor implements ExecutorInterface
```

Comment:
First: You mean AWS Lambda? It's an unlikely assumption we will only execute a command on lambda and all other processing will be done on another machine. The more likely point of decoupling is the RabbitMQ queue, i.e. the full crawler. In this case, no decoupling of the command execution is needed.

Second on testing: usually, the recommended textbook advice is to create two classes, one to obtain the data (in this case by calling a function, in other cases it would be from an API) and the other one to use this data in some way. The first class is not unit tested, but covered by integration tests and for the purposes of unit tests assumed to be working. The second class (which does the processing and contains the relevant custom code) is the one tested by unit tests.

Now, we have a bit of a mixture between getting data and processing it in the crawler classes. That's fine, as it leads to efficient and readable code. Testing this setup is also easy: create a shell script called "git", which you inject into the `PATH` environment variable before running tests. Then you can let the shell script return your golden output, just like git would, and be done with it. No large mocking framework or overloaded class hierarchy needed.